File: hw/aymacana/u3/hw27GUIsOperation/Contacts_JS/controller/ContactController.js
// controller/SoccerTeamController.js
import SimpleCrud from './SimpleCrud.js';
import Contact from '../model/Contact.js';
import IdGenerator from '../utils/idGenerator.js';

class ContactController {
    constructor() {
        this.crud = new SimpleCrud('Contacts');
    }

    // CREATE
    async addTeam(name, phone, email, addres) {
        try {
            // Generar ID autoincremental
            const nextId = await IdGenerator.getNextId('Contacts');
            
            // Crear equipo
            const contact = new Contact(
                nextId,
                name,
                phone,
                email,
                addres
            );
            
            // Guardar en MongoDB
            const doc = contact.toDocument();
            const newId = await this.crud.create(doc);
            
            return {
                success: true,
                message: ` Contacto agregado con ID: ${newId}`,
                data: contact
            };
            
        } catch (error) {
            return {
                success: false,
                message: ` Error al agregar contacto: ${error.message}`,
                data: null
            };
        }
    }
}

export default ContactController;

File: hw/aymacana/u3/hw27GUIsOperation/Contacts_PY/controller/contact_controller.py
# controller/soccer_team_controller.py
from controller.simple_crud import SimpleCrud
from model.contact import Contact
from utils.id_generator import IdGenerator

class ContactController:
    def __init__(self):
        self.crud = SimpleCrud("Contacts")
    
    # CREATE
    def add_contact(self, name, phone, email, addres):
        try:
            # Generar ID autoincremental
            next_id = IdGenerator.get_next_id("Contacts")
            
            # Crear equipo
            contact = Contact(
                next_id,
                name,
                phone,
                email,
                addres
            )
            
            # Guardar en MongoDB
            contact_dict = contact.to_dict()
            new_id = self.crud.create(contact_dict)
            
            return {
                "success": True,
                "message": f"✅ Contacto agregado con ID: {new_id}",
                "data": contact
            }
            
        except Exception as e:
            return {
                "success": False,
                "message": f"❌ Error al agregar equipo: {str(e)}",
                "data": None
            }
    
    # Check if team exists
    def team_exists(self, team_id):
        return self.crud.exists(team_id)